packages/nimble-components/src/table/types.ts

```diff
@@ -211,4 +214,43 @@ export interface TableRowState<TData extends TableRecord = TableRecord> {
     groupColumn?: TableColumn;
     isParentRow: boolean;
     isLoadingChildren: boolean;
+    dataIndex?: number;
+}
+
+/**
+ * @internal
+ * Table keyboard focus types
+ */
+export const TableFocusType = {
+    none: 'none',
+    columnHeader: 'columnHeader',
+    headerActions: 'headerActions',
+    row: 'row',
+    rowSelectionCheckbox: 'rowSelectionCheckbox',
+    cell: 'cell',
+    cellActionMenu: 'cellActionMenu',
+    cellContent: 'cellContent'
+} as const;
+export type TableFocusType =
+    (typeof TableFocusType)[keyof typeof TableFocusType];
+
+/**
+ * @internal
+ * Focusable elements of a table row
+ */
+export interface TableRowFocusableElements {
+    selectionCheckbox?: Checkbox;
+    cells: {
+        cell: TableCell,
```

Comment:
Is it worth making this a bit more robust, and allowing individual cells to have multiple focusable elements? I suspect that it's probably not a big refactor to enable this in the future if need be.

Reply:
That should be supported now (though as you said there's not a use case / tests that cover it yet).

Any focusable elements for a column type should be exposed via [cellView.tabbableElements](https://github.com/ni/nimble/blob/e807d833f7b5758642806a7053896ca81aa9eed7/packages/nimble-components/src/table-column/base/cell-view/index.ts#L36) (an array, so there can be multiple), and the keyboard nav code supports navigating between those with Tab.